conversion/conversion.go

```diff
@@ -387,11 +409,20 @@ func dataFromDatabaseForDataflowMigration(targetProfile profiles.TargetProfile,
 		if err != nil {
 			return common.TaskResult[*profiles.DataShard]{Result: p, Err: err}
 		}
+
+		// Fetch and store the GCS bucket associated with the datastream
+		dsClient := getDatastreamClient(ctx)
+		gcsBucket, _, fetchGcsErr := streaming.FetchTargetBucketAndPath(ctx, dsClient, targetProfile.Conn.Sp.Project, streamingCfg.DatastreamCfg.DestinationConnectionConfig)
+		if fetchGcsErr != nil {
+			logger.Log.Error(fmt.Sprintf("Could not fetch GCS Bucket for Shard %s hence Monitoring Dashboard will not contain Metrics for the gcs bucket\n error=%v\n", p.DataShardId, fetchGcsErr))
+		}
```

Comment:
In my own experimentation I recently realised that the zap logger (that we use in HB), prints the whole stack trace when logging at the `ERROR` level.  This results in nasty looking stack traces when an error is encountered by the customer. 
Because we allow customers to set log-levels, and they probably don't want to see these stack traces unless they are running at `DEBUG` level, I would suggest something like the following - 

1. Log all errors at the `INFO` level. This will just print the error for the customer and not HB internal stack trace.
2. Add a second log below the `INFO` log which prints errr stack trace only if the log level is set to `DEBUG`.

I understand that this might require some tweaking (for example, make `DEBUG` behave the same way as `ERROR` ?) which might be out of scope of this PR and more relevant to "fix logging in HB" that we already have in our plate. Just noting this here down for posterity.

Mentioning #350 here to link back to this comment whenever we get to it. @bharadwaj-aditya something to think about..